LanguageFeatures/Augmentation-libraries/augmented_expression_A03_t28.dart

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The exact result of an `augmented` expression depends on what is
+/// being augmented, but it generally follows the same rules as any normal
+/// identifier:
+/// ...
+/// - Augmenting fields: Within an augmenting field, augmented can only be used
+///   in an initializer expression, and refers to the original field's
+///   initializer expression, which is immediately evaluated.
+///
+/// It is a compile-time error to use augmented in an augmenting field's
+/// initializer if the member being augmented is not a field with an initializer
+///
+/// @description Checks that within an augmenting field `augmented` invokes the
+/// original field's initializer expression. Test a local function.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+import augment 'augmented_expression_A03_t28_lib.dart';
+import '../../Utils/expect.dart';
+
+String augmented() => "Top-level augmented(), shouldn't be invoked";
+
+String foo() => "Original";
```

Comment:
I think it would be useful for `foo` to have a side effect (incrementing a counter or something like that), such that we can see that it evaluates the initializing expression. This would show that we aren't just reading the value of a storage location, we are actually evaluating the initializing expression.

But this means that `foo` should not be a tear-off, it should be a getter that returns a function object (or we could just make it a getter and then omit `()` from all the occurrences of `augmented` below).

We might want to have at least a couple of cases where `augmented` or `augmented()` is invoked more than once, again to detect that a non-first evaluation is still an evaluation of the expression, not a stored value.

Reply:
Thanks!